cogito/protocol.go

```diff
@@ -262,18 +262,13 @@ func (src *Source) Validate() error {
 		if src.Repo == "" {
 			mandatory = append(mandatory, "repo")
 		}
-
-		if !src.GitHubApp.IsZero() {
-			if src.GitHubApp.ClientId == "" {
-				mandatory = append(mandatory, "github_app.client_id")
-			}
-			if src.GitHubApp.InstallationId == 0 {
-				mandatory = append(mandatory, "github_app.installation_id")
-			}
-			if src.GitHubApp.PrivateKey == "" {
-				mandatory = append(mandatory, "github_app.private_key")
+		// if access token is not set; we are using GitHub app. Validate it.
+		if src.AccessToken == "" {
```

Comment:
I find this way of testing confusing. It seems to me that the previous way (deleted line 266) was more straightforward. What was the reason to change the logic for the `if` ?

Reply:
At that point, we know that `src.AccessToken != ""` xor `src.GitHubApp != github.GitHubApp{}` is true so I agree with Marco that doing the corresponding direct test would be better than the contrapositive.
```suggestion
		// if access token is not set; we are using GitHub app. Validate it.
		if src.GitHubApp != github.GitHubApp{} {
```

Reply:
Done as suggested. I keep using the `IsZero()` method..